fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/service/LoanWritePlatformServiceJpaRepositoryImpl.java

```diff
@@ -295,6 +303,139 @@ public CommandProcessingResult disburseLoan(Long loanId, JsonCommand command, Bo
+                }
+
+                BigDecimal loanOutstanding = this.loanReadPlatformService
+                        .retrieveLoanPrePaymentTemplate(LoanTransactionType.REPAYMENT, loanIdToClose, actualDisbursementDate).getAmount();
+                final BigDecimal firstDisbursalAmount = loan.getFirstDisbursalAmount();
+                if (loanOutstanding.compareTo(firstDisbursalAmount) > 0) {
+                    throw new GeneralPlatformDomainRuleException("error.msg.loan.amount.less.than.outstanding.of.loan.to.be.closed",
+                            "Topup loan amount should be greater than outstanding amount of loan to be closed.");
+                }
+            }
+            if (loan.getRepaymentScheduleInstallments().isEmpty()) {
+                /*
+                 * If no schedule, generate one (applicable to non-tranche multi-disbursal loans)
+                 */
+                recalculateSchedule = true;
+            }
+
+            regenerateScheduleOnDisbursement(command, loan, recalculateSchedule, scheduleGeneratorDTO, nextPossibleRepaymentDate,
+                    rescheduledRepaymentDate);
+            boolean downPaymentEnabled = loan.repaymentScheduleDetail().isEnableDownPayment();
+            if (loan.repaymentScheduleDetail().isInterestRecalculationEnabled() || downPaymentEnabled) {
+                createAndSaveLoanScheduleArchive(loan, scheduleGeneratorDTO);
+            }
+        }
+
+        final PortfolioAccountData portfolioAccountData = this.accountAssociationsReadPlatformService
+                .retriveLoanLinkedAssociation(loan.getId());
+        if (portfolioAccountData == null) {
+            final String errorMessage = "Disburse Loan with id:" + loan.getId() + " requires linked savings account for payment";
+            throw new LinkedAccountRequiredException("loan.disburse.to.savings", errorMessage, loan.getId());
+        }
+
+        InteropIdentifier identifier1 = interopService.getIdentifierByAccountId(portfolioAccountData.getId());
+        String payerIdentifierType = InteropIdentifierType.ACCOUNT_ID.toString();
+        String payerIdentifierValue = loan.getAccountNumber();
+        String payeeIdentifierType = identifier1.getType().toString();
+        String payeeIdentifierValue = identifier1.getValue();
+        String currency = amountBeforeAdjust.getCurrencyCode();
+        String amount = Integer.toString(amountBeforeAdjust.getAmount().intValue());
+        SdkDisbursalService sdkDisbursalService = new SdkDisbursalServiceImpl();
+        try {
+            String id = sdkDisbursalService.processDisbursal(payerIdentifierType, payerIdentifierValue, payeeIdentifierType,
+                    payeeIdentifierValue, amount, currency);
+            logger.info("Payment hub transaction started with transaction id: " + id);
+        } catch (Exception e) {
+            logger.error(e.getMessage());
```

Comment:
Please dont just silently swallow the outcome. How can the `disbursement` action be successful when the `sdkDisbursalService` failed for any reason...

Please review this logic!